cit3.0-web/src/contexts/authStateContext.js

```diff
@@ -15,13 +18,28 @@ const AuthStateContextProvider = ({ children }) => {
   const keycloakReady = useSelector((state) => state.keycloakReady);
   const dispatch = useDispatch();
 
+  // Handle modal on error
+  const configuration = useConfiguration();
+  const [show, setShow] = React.useState(false);
+  const handleShow = () => setShow(true);
+  const handleClose = () => {
+    setShow(false);
+    keycloak.obj.logout({
+      redirectUri: `${configuration.baseUrl}`,
+    });
+  };
+
   React.useEffect(() => {
     if (keycloak.obj && keycloak.obj.authenticated) {
       let idp = "";
       const loadUserInfo = () => {
         keycloak.obj
           .loadUserInfo()
           .then((user) => {
+            if (!user.email) {
+              handleShow();
```

Comment:
We should probably add a method to logout when leaving the page too. 

Such use case could be I see the message, and instead of closing the modal, I exit out of the tab. useEffect cleanup might work?